API/CopilotService/__init__.py

```diff
@@ -48,23 +49,36 @@ def main(req: func.HttpRequest, context: func.Context) -> func.HttpResponse:
     try:
         result = []
         if service_type == ServiceType.KNOWLEDGE_SEARCH:
-            result = knowledge_search(question, top_num)
-            if len(result) == 0 or not pass_verification(context, question, result):
-                result = []
+            try:
+                result = knowledge_search(question, top_num)
+                if len(result) == 0 or not pass_verification(context, question, result):
+                    result = []
+            except HttpResponseError as e:
+                logger.error('Response Status 500: Error from knowledge search: \n%s', e, e)
```

Comment:
>  e, e)

Why do we need to repeat the `e`

Reply:
The second `e` arg is for exec_info, which could help log the trace of the error.

Reply:
Fix the error with `exc_info=e`